### PR TITLE
Elasticsearch for Service Broker

### PR DESCRIPTION
**Pull request by ndrest-amzn**
## Overview
Amazon Elasticsearch service capability for service broker

## Testing

Tested template for Prod and Development settings. E.g Single and 2/3 AZ deployments with single and multi data nodes and master nodes per recommend doc guidance. 
Includes specific elasticsearch encryption at rest and node to node encryption options.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: templates/elasticsearch/template.yaml [hunk 1]
```diff
@@ -0,0 +1,912 @@
...
+          AvailabilityZones: Auto
+          CidrBlocks: Auto
+          CidrSize: 27          
+          ESDomainName: Auto
+          NodeAvailabilityZones: 1          
+          NodeInstanceCount: 1          
+          DedicatedMasterInstanceCount: 0          
+          NodeToNodeEncryption: true
+          EncryptionAtRest: true
+      custom:
```
Comment (jaymccon) [inline templates/elasticsearch/template.yaml]: `Description` and `Cost` are required by CF

File: templates/elasticsearch/template.yaml [hunk 2]
```diff
@@ -0,0 +1,912 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: AWS Service Broker - Amazon Elasticsearch
```
Comment (jaymccon) [inline templates/elasticsearch/template.yaml]: can you add the template id `(qs-1poknmeof)` to the end of the description

Comment (ndrest-amzn) [reply templates/elasticsearch/template.yaml]: Added ID.

File: templates/elasticsearch/template.yaml (hunk 1 above)
Comment (ndrest-amzn) [reply templates/elasticsearch/template.yaml]: Only see Description missing under 'custom:', so added that in. Cost is there. But let me know if an issue.

**Comment by jaymccon** [comment]
great stuff @ndrest-amzn! this is a much requested service, so I'm sure will be helpful to many

**Pull request merged**